### PR TITLE
Add a working Vagrantfile

### PR DESCRIPTION
**Pull request by rylev**
This gives us a development environment where one can build and run the kernel. The only issue I had was when trying to run the kernel I got an error from qemu `"Could not initialize SDL(No available video device) - exiting"`. In order to get it running I added `-curses` to the qemu command to bypass using SDL. That works, but I'm not sure it's ideal for the long term.

### CODE REVIEW

**Comment by steveklabnik** [comment]
/cc @ashleygwilliams who wanted to compare/contrast

Thanks @rylev !

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  # documentation for more information about their specific syntax and use.
+   config.vm.provision "shell", inline: <<-SHELL
+      sudo apt-get update
+      sudo apt-get install nasm -y
+      sudo apt-get install xorriso -y
+      sudo apt-get install git -y
+      sudo apt-get install vim -y
+      sudo apt-get install -y qemu
+      sudo apt-get install -y make
+      sudo apt-get install -y binutils
```
Comment (steveklabnik) [inline Vagrantfile]: could we make the -y consistent either before or after? If i had to pick I'd say before, but I don't care either way.

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      curl -sf https://raw.githubusercontent.com/brson/multirust/master/blastoff.sh | sh -s -- --yes
+      multirust default nightly-2015-11-19
```
Comment (steveklabnik) [inline Vagrantfile]: does this work? I know @ashleygwilliams had tried it, and it seemed like there were still issues. It should though?

File: Vagrantfile (hunk 1 above)
Comment (rylev) [inline Vagrantfile]: I vote before too. I'll make the change.

File: Vagrantfile (hunk 2 above)
Comment (rylev) [inline Vagrantfile]: Yea I didn't seem to have any issues with it.

**Comment by steveklabnik** [comment]
Thanks a ton!

**Pull request merged**